tests/generics/monomorphizer.test.ts

```diff
@@ -0,0 +1,130 @@
+import { expect } from 'chai';
+import { Program, Type } from '../../ast';
+import { monomorphizeProgram } from '../../monomorphizer';
+
+describe('monomorphizeProgram(program) function', () => {
+    it('monomorphizes a generic class in a program with two var inits with generic type int', () => {
+        const tast : Program<Type> = { a: { tag: 'none' }, inits: [{name: 'c1', type: { tag: 'class', name: 'C', params: [ { tag: 'number' } ] }, 
```

Comment:
Whoa nice. Not great to have too many of these, but some is cool.

These will be a pain to maintain when AST changes.

Reply:
I realize I have way too many tests like this that actually use the AST for the Type-Checker..... It was the only way to test independently before the parser and monomorphizer were done. Hopefully the remaining AST changes from this point own will be trivial.